application.js

```diff
@@ -104,6 +104,8 @@ const Application = GObject.registerClass(
 
             this.add_action(this.window.toggle_action);
             this.add_action(this.window.hide_action);
+            this.add_action(this.window.toggle_maximize_action);
+            this.add_action(this.window.toggle_unmaximize_action);
```

Comment:
This copies `win.*` actions to `app.*`. Shouldn't be necessary.

Reply:
If I'm not mistaken, this is necessary only for actions that you want to work globally - i. e. when terminal window isn't active. IMO it's not needed for maximize/unmaximize